csrc/expr_simplifier.cpp

```diff
@@ -2791,6 +2791,9 @@ Val* simplifyExpr(
   std::unique_ptr<std::unordered_set<std::string>> disabled_passes = nullptr;
   if (isOptionDisabled(DisableOption::ExprSimplify)) {
     const auto& v = getDisableOptionArguments(DisableOption::ExprSimplify);
+    if (v.empty()) {
```

Comment:
Currently, even if `expr_simplify` is disabled, `flatten` and `unflatten` are done anyway. This makes sure that nothing is really done. Confirmed with @zasdfgbnm